### PR TITLE
numaresources-operator: add e2e tests

### PR DESCRIPTION
**Pull request by mmirecki**
Add e2e tests dependant on olm installation

### CODE REVIEW

**Comment by fromanirh** [comment]
@mmirecki fixed the ci-e2e failure: https://github.com/openshift-kni/numaresources-operator/pull/21

File: ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml [hunk 1]
```diff
@@ -47,6 +47,27 @@ tests:
...
   container:
     from: src
+- as: ci-e2e
+  steps:
+    cluster_profile: aws
+    env:
+      OO_CHANNEL: alpha
+      OO_INSTALL_NAMESPACE: numaresources-operator
+      OO_PACKAGE: numaresources-operator
+      OO_TARGET_NAMESPACES: '!all'
```
Comment (cynepco3hahue) [inline ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml:57]: probably it is not correct, and we should not install pao under all namespace, but we should first fix it under the numaresource-operator

File: ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml [hunk 2]
```diff
@@ -47,6 +47,27 @@ tests:
...
+    test:
+    - as: e2e-test
+      cli: latest
+      commands: make test-e2e
+      dependencies:
+      - env: OO_INDEX
```
Comment (cynepco3hahue) [inline ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml:63]: I think we will need to add some additional ENV variables, see - https://github.com/openshift-kni/numaresources-operator/blob/939709fee7a1c4b58463cb66e353f723139c2c90/.github/workflows/e2e.yml#L21

Comment (mmirecki) [reply ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml:63]: added:
      - env: E2E_NAMESPACE_NAME
        name: numaresources
      - env: E2E_TOPOLOGY_MANAGER_POLICY
        name: SingleNUMANodeContainerLevel

File: ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml (hunk 1 above)
Comment (mmirecki) [reply ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml]: changed to '!install'

**Comment by fromanirh** [comment]
@mmirecki this is the last PR we think it's needed on the operator side: /github.com/openshift-kni/numaresources-operator/pull/50
it's merged now, so you should be able to have green lane from now on.

**Comment by fromanirh** [comment]
@Tal-or could you please check this PR is up do date with the recent e2e fixes we merged on NROP?

**Comment by cynepco3hahue** [comment]
I think the problem is with ENV variables, we should add it differently because I can see
```
 step ci-e2e failed: "ci-e2e" test steps failed: [could not determine image pull spec for image numaresources on step e2e-test, could not determine image pull spec for image SingleNUMANodeContainerLevel on step e2e-test] 
```

**Comment by cynepco3hahue** [comment]
Probably it should not be under dependencies, see - https://github.com/openshift/release/blob/69979cbfefe9effd91a9facf17cad60f54e8d150/ci-operator/config/openshift/origin-aggregated-logging/openshift-origin-aggregated-logging-tech-preview.yaml#L143

File: ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml (hunk 2 above)
Comment (Tal-or) [reply ci-operator/config/openshift-kni/numaresources-operator/openshift-kni-numaresources-operator-main.yaml:63]: These env variables are used for other purposes: 
https://docs.ci.openshift.org/docs/architecture/ci-operator/#referring-to-images-in-tests 

You can remove them.
I added them here:

**Comment by Tal-or** [comment]
Just post my answer here:
https://github.com/openshift/release/pull/23898/commits/1f58cb6305606acd054bf3d85a0a16673f55d697#r768542732

**Comment by Tal-or** [comment]
> Probably it should not be under dependencies, see -
> 
> https://github.com/openshift/release/blob/69979cbfefe9effd91a9facf17cad60f54e8d150/ci-operator/config/openshift/origin-aggregated-logging/openshift-origin-aggregated-logging-tech-preview.yaml#L143

We don't need this env at all.
I already added them inside the script itself:
https://github.com/openshift-kni/numaresources-operator/blob/main/hack/run-test-e2e.sh#L26
It's more flexible this way.
We can change them via our repo and not from here

**Comment by Tal-or** [comment]
> I think the problem is with ENV variables, we should add it differently because I can see
> 
> ```
>  step ci-e2e failed: "ci-e2e" test steps failed: [could not determine image pull spec for image numaresources on step e2e-test, could not determine image pull spec for image SingleNUMANodeContainerLevel on step e2e-test] 
> ```

it refer to images only:
https://docs.ci.openshift.org/docs/architecture/ci-operator/#referring-to-images-in-tests

**Comment by mmirecki** [comment]
> > I think the problem is with ENV variables, we should add it differently because I can see
> > ```
> >  step ci-e2e failed: "ci-e2e" test steps failed: [could not determine image pull spec for image numaresources on step e2e-test, could not determine image pull spec for image SingleNUMANodeContainerLevel on step e2e-test] 
> > ```
> 
> it refers to images only: https://docs.ci.openshift.org/docs/architecture/ci-operator/#referring-to-images-in-tests

Moving E2E_NAMESPACE_NAME to env, discarding E2E_TOPOLOGY_MANAGER_POLICY

**Comment by Tal-or** [comment]
The logs says: ```parameter \"E2E_NAMESPACE_NAME\" is overridden in [test/ci-e2e] but not declared in any step```
Although I'm not sure what are the steps that the ```E2E_NAMESPACE_NAME``` suppose to be declared in,  we can just call it this way:
```commands: E2E_NAMESPACE_NAME=numaresources make test-e2e```

**Comment by fromanirh** [comment]
thanks @mmirecki @Tal-or  @cynepco3hahue for helping reaching this milestone!

**Pull request merged**